src/vehicle.cpp

```diff
@@ -2063,41 +2068,18 @@ int vehicle::next_part_to_open( int p, bool outside ) const
     return -1;
 }
 
-/**
- * Returns all parts in the vehicle with the given flag, optionally checking
- * to only return unbroken parts.
- * If performance becomes an issue, certain lists (such as wheels) could be
- * cached and fast-returned here, but this is currently linear-time with
- * respect to the number of parts in the vehicle.
- * @param feature The flag (such as "WHEEL" or "CONE_LIGHT") to find.
- * @param unbroken true if only unbroken parts should be returned, false to
- *        return all matching parts.
- * @return A list of indices to all the parts with the specified feature.
- */
-std::vector<int> vehicle::all_parts_with_feature( const std::string &feature,
-        bool const unbroken ) const
+vehicle_part_with_feature_range<std::string> vehicle::parts_with_feature( std::string feature,
+        const bool unbroken ) const
 {
-    std::vector<int> parts_found;
-    for( size_t part_index = 0; part_index < parts.size(); ++part_index ) {
-        if( part_info( part_index ).has_flag( feature ) &&
-            ( !unbroken || !parts[ part_index ].is_broken() ) ) {
-            parts_found.push_back( part_index );
-        }
-    }
-    return parts_found;
+    return vehicle_part_with_feature_range<std::string>( const_cast<vehicle &>( *this ),
+            std::move( feature ), unbroken );
```

Comment:
What is this std:move accomplishing?

Reply:
Not much, I agree. But I figured this function is usually called with a string literal, so the `std::string` object is created just for this function call. Forwarding it via `std::move` means it's not copied but moved into the `vehicle_part_with_feature_range` (note that the function take the `std::string` as value now). So we have a few movements, instead of copies.